apps/settings/js/panels/apn_editor/apn_editor_session.js

```diff
@@ -41,9 +41,10 @@ define(function() {
       var promises = [];
       var newApnSetting = this._exportApnSetting(this._inputElements);
       newApnSetting.types.slice().forEach(function(type) {
-        newApnSetting.types = [type];
+        var settingClone = ApnUtils.clone(newApnSetting);
```

Comment:
:100:  
Please add some note here to help people know we should clone object here or the promise push will always take the last newApnSetting.types value